app/src/db/bulk_ops.py

```diff
@@ -0,0 +1,150 @@
+"""Bulk database operations for performance.
+
+Provides a bulk_upsert function
+"""
```

Comment:
maybe update doc for this module or for the bulk_upsert function that this is postgres + psycopg specific